src/storage/secondary/rowset/rowset_iterator.rs

```diff
@@ -450,4 +543,293 @@ mod tests {
             unreachable!()
         }
     }
+
+    #[tokio::test]
+    async fn test_rowset_iterator_with_range_filter() {
+        {
+            let tempdir = tempfile::tempdir().unwrap();
+            let rowset = Arc::new(helper_build_rowset_with_first_key_recorded(&tempdir).await);
+            let start_keys = vec![DataValue::Int32(180)];
+            let end_keys = vec![DataValue::Int32(195)];
+            let mut it = rowset
+                .iter(
+                    vec![
+                        StorageColumnRef::Idx(0),
+                        StorageColumnRef::Idx(1),
+                        StorageColumnRef::Idx(2),
+                    ]
+                    .into(),
+                    vec![],
+                    ColumnSeekPosition::RowId(168),
+                    None,
```

Comment:
Filter condition should include the range info. Here we can construct an expression manually -- `180 < InputRef(0) < 195`.

Reply:
May fix in later PRs.